qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,511 @@
+    for term in fer_op.to_list():
+        _add_edges_for_term(edge_matrix, operator_string(term))
+    return edge_matrix
+
+
+def _add_one_edge(edge_matrix, i: int, j: int):
+    """
+    Add an edge from lesser index to greater. This maintains the upper triangular structure.
+    """
+    if i < j:
+        edge_matrix[i, j] = True
+    elif j < i:
+        edge_matrix[j, i] = True
+    else:
+        raise ValueError("expecting i != j")
+
+
+def _add_edges_for_term(edge_matrix, term_str: str):
+    """
+    Add one, two, or no edges to `edge_matrix` as dictated by the operator `term_str`.
+    """
+    (n_number, n_raise, n_lower), facs = _unpack_term(term_str)
+    ttype = _interaction_type(n_number, n_raise, n_lower)
+    # For 'excitation' and 'number_excitation', create and edge between the `+` and `-`.
+    if ttype in ("excitation", "number_excitation"):
+        inds = [i for (i, c) in facs if c in "+-"]
+        if len(inds) != 2:
+            raise ValueError("wrong number or raising and lowering")
+        _add_one_edge(edge_matrix, *inds)
+    # For `double_excitation` create an edge between the two `+`s and edge between the two `-`s.
+    elif ttype == "double_excitation":
+        raise_inds = [i for (i, c) in facs if c == "+"]
+        lower_inds = [i for (i, c) in facs if c == "-"]
+        _add_one_edge(edge_matrix, *raise_inds)
+        _add_one_edge(edge_matrix, *lower_inds)
+
+
+def bksf_edge_list_fermionic_op(fer_op_qn: FermionicOp):
+    """
+    Construct edge list required for the bksf algorithm.
+
+    Args:
+        fer_op_qn: the fermionic operator in the second quantized form
+
+    Returns:
+        numpy.ndarray: edge_list, a 2xE matrix, where E is total number of edge
+                        and each pair denotes (from, to)
+    """
+    edge_matrix = _get_adjacency_matrix(fer_op_qn)
+    edge_list_as_2d_array = np.asarray(np.nonzero(edge_matrix))
+    return edge_list_as_2d_array
+
+
+def edge_operator_aij(edge_list: List, i: int, j: int):
+    """Calculate the edge operator A_ij.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list (numpy.ndarray): a 2xE matrix, where E is total number of edge
+                                    and each pair denotes (from, to)
+        i (int): specifying the edge operator A
+        j (int): specifying the edge operator A
+
+    Returns:
+        Pauli: qubit operator
+    """
+    v = np.zeros(edge_list.shape[1])
+    w = np.zeros(edge_list.shape[1])
+
+    position_ij = -1
+    qubit_position_i = np.asarray(np.where(edge_list == i))
+
+    for edge_index in range(edge_list.shape[1]):
+        if set((i, j)) == set(edge_list[:, edge_index]):
+            position_ij = edge_index
+            break
+
+    w[position_ij] = 1
+
+    for edge_index in range(qubit_position_i.shape[1]):
+        i_i, j_j = qubit_position_i[:, edge_index]
+        i_i = 1 if i_i == 0 else 0  # int(not(i_i))
```

Comment:
Shall we use the commented alternative?

Reply:
IIRC I copied the alternative from older code and replaced it by the conditional, changing the old code to a comment. But, I think the old way is fine too. I have changed the code to use `int(not(i_i))`.